sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs

```diff
@@ -134,14 +153,38 @@ public partial interface IDynamoDBContext : IDisposable
 
         #region Batch Write
 
+        /// <summary>
+        /// Creates a strongly-typed BatchWrite object, allowing
+        /// a batch-write operation against DynamoDB.
+        /// </summary>
+        /// <typeparam name="T">Type of objects to write</typeparam>
+        /// <returns>Empty strongly-typed BatchWrite object</returns>
+        BatchWrite<T> CreateBatchWrite<T>();
+
         /// <summary>
         /// Creates a strongly-typed BatchWrite object, allowing
         /// a batch-write operation against DynamoDB.
         /// </summary>
         /// <typeparam name="T">Type of objects to write</typeparam>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
         /// <returns>Empty strongly-typed BatchWrite object</returns>
-        BatchWrite<T> CreateBatchWrite<T>(DynamoDBOperationConfig operationConfig = null);
+        [Obsolete("Use the CreateBatchWrite overload that takes BatchWriteConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to BatchWrite.")]
+
+        BatchWrite<T> CreateBatchWrite<T>(DynamoDBOperationConfig operationConfig);
```

Comment:
why did you change the signature of the method?

Reply:
See https://github.com/aws/aws-sdk-net/pull/3376#discussion_r1672274431